src/rgw/rgw_op.cc

```diff
@@ -1424,10 +1425,9 @@ void RGWCreateBucket::execute()
     if (ret < 0) {
       ldout(s->cct, 0) << "WARNING: failed to unlink bucket: ret=" << ret << dendl;
     }
-  }
-
-  if (ret == -EEXIST)
+  } else if (ret == -EEXIST || (ret == 0 && existed)) {
```

Comment:
It looks like _cls_user_set_buckets_info()_ and thus _rgw_link_bucket()_ do not return EEXIST. Moreover, I cannot find anything claiming they should do that.